File: backend/app/modules/customer/models/customer.py
from sqlalchemy import Index, String, text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database.base import Base
from app.core.database.mixins.timestamp import TimestampMixin
from app.core.database.mixins.uuid import UUIDMixin


class Customer(
    UUIDMixin,
    TimestampMixin,
    Base,
):
    __tablename__ = "customer"

    __table_args__ = (
        Index(
            "uq_customer_document_id",
            "document_id",
            unique=True,
            postgresql_where=text("document_id IS NOT NULL"),
        ),
    )

    name: Mapped[str] = mapped_column(
        String(150),
        nullable=False,
        index=True,
    )

    document_id: Mapped[str | None] = mapped_column(
        String(20),
        nullable=True,
    )

    phone: Mapped[str | None] = mapped_column(
        String(30),
        nullable=True,
    )

    email: Mapped[str | None] = mapped_column(
        String(255),
        nullable=True,
    )

    address: Mapped[str | None] = mapped_column(
        String(255),
        nullable=True,
    )

    notes: Mapped[str | None] = mapped_column(
        String(500),
        nullable=True,
    )

    is_active: Mapped[bool] = mapped_column(
        nullable=False,
        default=True,
    )

    is_default_customer: Mapped[bool] = mapped_column(
        nullable=False,
        default=False,
    )

File: frontend/src/features/customers/types/customer.ts
// Mirrors backend/app/modules/customer/schemas/customer_*.py field-by-field.

export interface Customer {
  id: string
  name: string
  document_id: string | null
  phone: string | null
  email: string | null
  address: string | null
  notes: string | null
  is_active: boolean
  is_default_customer: boolean
  created_at: string
  updated_at: string
}

export interface CustomerCreatePayload {
  name: string
  document_id: string | null
  phone: string | null
  email: string | null
  address: string | null
  notes: string | null
}

export type CustomerUpdatePayload = Partial<CustomerCreatePayload>
